### PR TITLE
Rebalancing Manager for ethDai rebalancing set.

### PR DESCRIPTION
**Pull request by bweick**

### CODE REVIEW

File: contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+     * @param  _ethPriceFeedAddress     The address of ETH medianize
+     * @param  _daiAddress              The address of the Dai contract
+     * @param  _ethAddress              The address of the wrapped ETH contract
+     * @param  _setTokenFactory         The address of the SetTokenFactory
+     * @param  _auctionLibrary          The address of auction price curve to use in rebalance
+     * @param  _auctionTimeToPivot      The amount of time until pivot reached in rebalance
+     * @param  _multipliers             Token multipliers used to determine allocation
+     * @param  _allocationBounds        Bounds to stop proposal if not enough deviation from expected allocation
+     */
+    constructor(
```
Comment (felix2feng) [inline contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol]: Can we DRY this up similar to how its done with `ModuleCoreState` and the modules?

File: contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+
+        emit LogManagerProposal(
+            ethPrice
+        );
+    }
+
+    /* ============ Internal ============ */
+    /*
+     * Query price feed for weth, return as uint256
+     *
```
Comment (felix2feng) [inline contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol]: javadocs?

File: contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+            _ethPrice,
+            currentSetNaturalUnit,
+            currentSetUnits[1],
+            ETH_DECIMALS
+        );
+
+        // Total dollar value of currentSet (in cents)
+        uint256 currentSetDollarAmount = daiDollarAmount.add(ethDollarAmount);
+
+        // Require that the allocation has changed more than 2% in order for a rebalance to be called
```
Comment (felix2feng) [inline contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol]: Not 2% threshold in this case?

File: contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol (hunk 1 above)
Comment (bweick) [reply contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol]: Gonna do this when we generalize it

File: contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+        );
+        
+        // Create static components array
+        address[] memory nextSetComponents = new address[](2);
+        nextSetComponents[0] = daiAddress;
+        nextSetComponents[1] = ethAddress;
+
+        // Create the nextSetToken contract that collateralized the Rebalancing Set Token once rebalance
+        // is finished
+        address nextSetAddress = ICore(coreAddress).create(
```
Comment (felix2feng) [inline contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol]: FYI - We've renamed `create` to `createSet`

Comment (felix2feng) [reply contracts/supplementary/rebalancing-manager/ETHDaiRebalancingManager.sol]: Also you might need to coerce the output to `address`

File: deployments/constants.ts [hunk 5]
```diff
@@ -6,14 +6,27 @@ export default {
...
+    PRICE: new BigNumber(3711),
+    FULL_TOKEN_UNITS: new BigNumber(10 ** 8),
+    MULTIPLIER: new BigNumber(1),
+  },
   DEFAULT_WBTC_UNIT: new BigNumber(1),
-  WETH_FULL_TOKEN_UNITS: new BigNumber(10 ** 18),
-  WETH_PRICE: new BigNumber(128),
-  WETH_MULTIPLIER: new BigNumber(1),
+
+  WETH: {
```
Comment (kermankohli) [inline deployments/constants.ts]: Nice touch!

File: deployments/test/5_rebalancing.spec.ts [hunk 6]
```diff
@@ -233,4 +250,214 @@ describe('Deployment: Rebalancing', () => {
...
+      const receivedSetName = await ethDaiRebalancingSetToken.methods.name().call();
+      expect(receivedSetName).toEqual('BTD ETHDai Set');
+    });
+
+    it('rebalanced set should have the correct set symbol', async () => {
+      const receivedSetSymbol = await ethDaiRebalancingSetToken.methods.symbol().call();
+      expect(receivedSetSymbol).toEqual('ETHDai');
+    });
+
+  });
```
Comment (kermankohli) [inline deployments/test/5_rebalancing.spec.ts]: This looks mint

Comment (bweick) [reply deployments/test/5_rebalancing.spec.ts]: Unsure what Aussie slang that means...but I'm assuming you want me to remove the extra spaces haha

**Pull request merged**